app/serializers/vet_payment_history_serializer.rb

```diff
@@ -10,11 +10,70 @@ def id
     nil
   end
 
+  def initialize(object, options = nil)
+    @formatted_payments = []
+    @returned_payments = []
+
+    process_all_payments(object[:payments][:payment]) if object[:payments].present?
+    super
+  end
+
   def payments
-    object[:payments]
+    @formatted_payments
   end
 
   def return_payments
-    object[:return_payments]
+    @returned_payments
+  end
+
+  private
+
+  def process_all_payments(all_payments)
+    all_payments.each do |payment|
+      if payment.dig(:return_payment, :check_trace_number).present?
+        process_return_payment(payment)
+      else
+        process_payment(payment)
+      end
+    end
+  end
+
+  def process_payment(payment)
+    @formatted_payments << {
+      pay_check_dt: payment[:payment_date],
+      pay_check_amount: ActiveSupport::NumberHelper.number_to_currency(payment[:payment_amount]),
+      pay_check_type: payment[:payment_type],
+      payment_method: get_payment_method(payment),
+      bank_name: payment.dig(:address_eft, :bank_name),
+      account_number: mask_account_number(payment[:address_eft])
+    }
+  end
+
+  def process_return_payment(payment)
+    @returned_payments << {
+      'returned_check_issue_dt': payment[:payment_date],
+      'returned_check_cancel_dt': payment[:return_payment][:return_date],
+      'returned_check_amount': ActiveSupport::NumberHelper.number_to_currency(payment[:payment_amount]),
+      'returned_check_number': payment[:return_payment][:check_trace_number],
+      'returned_check_type': payment[:payment_type],
+      'return_reason': payment[:return_payment][:return_reason]
```

Comment:
Do you have any defaults for example if an attribute is nil or doesn't exist e.g. payment[:return_payment][:return_reason] where return_reason is nil?

Reply:
I'm not sure if I'm understanding your question correctly, so I'll do my best to answer/explain, but please clarify if I'm not answering your question.

So before we enter this method, it is using the "dig" method to determine whether or not [:return_payment][:check_trace_number] exists.  Using the "dig" method here prevents any errors in case [:return_payment] is nil (which it should not be, but just in case).  

Once  within this method, any of the return_payment attributes can be empty or nil.  
'returned_check_issue_dt'
'returned_check_cancel_dt'
'returned_check_amount'
'returned_check_number'
'returned_check_type'
'return_reason'

The FE is checking for nil values as well and putting something like "---" in the case of nil attributes.  

Does this help address your concern?

Reply:
Ah, thank you. I overlooked the dig method 👍

Reply:
Thank you @LindseySaari!